### PR TITLE
HPCC-30921 Improve Graph Rendering Root

### PR DESCRIPTION
**Pull request by GordonSmith**
* Add breadcrumbs for graph lineage
* Don't render parent for select subgraphs
* Add status spinner with node count

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith looked good, just a couple questions & saw some commented out code in there

File: esp/src/src-react/components/Metrics.tsx [hunk 1]
```diff
@@ -318,17 +370,8 @@ export const Metrics: React.FunctionComponent<MetricsProps> = ({
                 metricGraphWidget.centerOnItem(selectedMetrics[selectedMetricsPtr + 1].name);
                 setSelectedMetricsPtr(selectedMetricsPtr + 1);
             }
-        },
-        {
-            key: "reset", text: nlsHPCC.Reset, iconProps: { iconName: "Undo" },
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply esp/src/src-react/components/Metrics.tsx:323]: Yes intentional - seemed to cause more confusion than anything useful.

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith just saw a couple lines of commented out code in some of the new additions.

File: esp/src/src-react/components/Metrics.tsx [hunk 2]
```diff
@@ -439,17 +510,41 @@ export const Metrics: React.FunctionComponent<MetricsProps> = ({
...
+    React.useEffect(() => {
         if (selectedMetrics) {
             updateScopesTable(selectedMetrics);
-            updateMetricGraph(selectedMetrics);
             updatePropsTable(selectedMetrics);
             updatePropsTable2(selectedMetrics);
+            updateLineage(selectedMetrics);
         }
-    }, [], [selectedMetrics]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply esp/src/src-react/components/Metrics.tsx:543]: I don't want this to be called when the updateXXX functions changed

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**